From b14441dd83c080c1dae3a34f4c825c68b2299c2c Mon Sep 17 00:00:00 2001
From: Benjamin C Meyer <ben@meyerhome.net>
Date: Tue, 12 Jun 2012 22:49:22 -0400
Subject: [PATCH 11/23] Work around Qt bug in QCompleter found in Qt 4.8.2

A model needs to be passed to the completer otherwise it crashes.

Issue: 957
---
 src/toolbarsearch.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git src/toolbarsearch.cpp src/toolbarsearch.cpp
index 5f14206..5907920 100644
--- src/toolbarsearch.cpp
+++ src/toolbarsearch.cpp
@@ -104,8 +104,7 @@ ToolbarSearch::ToolbarSearch(QWidget *parent)
     connect(openSearchManager(), SIGNAL(currentEngineChanged()),
             this, SLOT(currentEngineChanged()));
 
-    m_completer = new QCompleter(this);
-    m_completer->setModel(m_model);
+    m_completer = new QCompleter(m_model, this);
     m_completer->setCompletionMode(QCompleter::UnfilteredPopupCompletion);
     setCompleter(m_completer);
 
-- 
2.44.0

